subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1490,6 +1494,45 @@ static void vs_read_supported_features(struct net_buf *buf,
 	memset(&rp->features[0], 0x00, sizeof(rp->features));
 }
 
+#if defined(CONFIG_BT_CTLR_HCI_VS_EXT)
+static void vs_read_static_addrs(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_rp_vs_read_static_addrs *rp;
+
+	rp = cmd_complete(evt, sizeof(*rp));
+
+#if defined(CONFIG_SOC_FAMILY_NRF5)
+	/* Read address from nRF5-specific storage
+	 * Non-initialized FICR values default to 0xFF, skip if no address
+	 * present. Also if a public address lives in FICR, do not use in this
+	 * function.
+	 */
+	if (((NRF_FICR->DEVICEADDR[0] != UINT32_MAX) ||
+	    ((NRF_FICR->DEVICEADDR[1] & UINT16_MAX) != UINT16_MAX)) &&
+	      (NRF_FICR->DEVICEADDRTYPE & 0x01)) {
+
+		rp->num_addrs = 1;
+		sys_put_le32(NRF_FICR->DEVICEADDR[0], &rp->a[0].bdaddr.val[0]);
+		sys_put_le16(NRF_FICR->DEVICEADDR[1], &rp->a[0].bdaddr.val[4]);
+		/* The FICR value is a just a random number, with no knowledge
+		 * of the Bluetooth Specification requirements for random
+		 * static addresses.
+		 */
+		BT_ADDR_SET_STATIC(&rp->a[0].bdaddr);
+
+		/* Mark IR as invalid */
+		memset(rp->a[0].ir, 0x00, sizeof(rp->a[0].ir));
+
+		rp->status = 0x00;
+
+		return;
+	}
+#endif /* CONFIG_SOC_FAMILY_NRF5 */
+
+	rp->status = BT_HCI_ERR_UNSPECIFIED;
```

Comment:
Where is num_addrs handled for the non-nRF5x case. There is really no point in returning an error here. Just return success and an num_addrs = 0. That said, I think the cmd_complete and sizeof(*rp) are not flexible here. You need to able to return num_addrs = 0 and num_addrs = 1.